### PR TITLE
Re-implement a few model types in Swift

### PR DESCRIPTION
**Pull request by crazytonyli**
### Description

What says in the title 😄 

### Testing Details

This PR should be low risk, as it only changed type declaration.

- [x] Please check here if your pull request includes additional test coverage.
- [x] I have considered updating the `version` in the `.podspec` file.

### CODE REVIEW

File: WordPressKit/RemoteComment.swift [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import Foundation
+
+@objcMembers public class RemoteComment: NSObject {
```
Comment (jkmassel) [inline WordPressKit/RemoteComment.swift:3]: TIL `@objcMembers`!

File: WordPressKit/RemoteSourcePostAttribution.swift [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    public var authorName: String?
+    public var authorURL: String?
+    public var blogName: String?
+    public var blogURL: String?
+    public var avatarURL: String?
+    public var blogID: NSNumber?
+    public var postID: NSNumber?
+    public var likeCount: NSNumber?
+    public var commentCount: NSNumber?
+    public var taxonomies: NSArray?
```
Comment (jkmassel) [inline WordPressKit/RemoteSourcePostAttribution.swift]: Does Swift know about the underlying type here? If so, can we use a Swift collection?

**Review by jkmassel — CHANGES_REQUESTED** [review]

File: WordPressKit/RemoteSourcePostAttribution.swift (hunk 2 above)
Comment (crazytonyli) [reply WordPressKit/RemoteSourcePostAttribution.swift]: Addressed in https://github.com/wordpress-mobile/WordPressKit-iOS/pull/560/commits/d2d11251072ac48a8df2c5fe09a05b54b5a8980b

**Review by mokagio — APPROVED** [review]
Approving to unblock as @jkmassel 's concern about `NSArray` [have been addressed](https://github.com/wordpress-mobile/WordPressKit-iOS/pull/560#discussion_r1048961185).

**Comment by crazytonyli** [comment]
I've changed the base branch to the dedicated Swift migration branch, see https://github.com/wordpress-mobile/WordPressKit-iOS/pull/562 for more context.

**Pull request merged**